backend/dataall/modules/worksheets/aws/s3_client.py

```diff
@@ -0,0 +1,73 @@
+import boto3
+from botocore.config import Config
+
+from botocore.exceptions import ClientError
+import logging
+from dataall.base.db.exceptions import AWSResourceNotFound
+from dataall.base.aws.sts import SessionHelper
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from dataall.core.environment.db.environment_models import Environment
+    try:
+        from mypy_boto3_s3 import S3Client as S3ClientType
+    except ImportError:
+        S3ClientType = None
+
+log = logging.getLogger(__name__)
+
+
+class S3Client:
+
```

Comment:
just curious: why are you defining type hints with quotes? (e.g. lin 23: env: 'Environment'):)

Reply:
There are two type of type hints available in python.

1. Backward Reference `env: Environment`
2. Forward Reference `env: 'Environment'`

The major difference between two is
**Backward Reference** requires Type hints to be present in the current local or global scope but
**Forward Reference** resolves type hints run time and does not requires type hints to be present in scope.